core/domain/topic_domain.py

```diff
@@ -214,9 +220,17 @@ def delete_story(self, story_id):
         Args:
             story_id: str. The story id to remove from the list.
 
+        Raises:
+            Exception. The story_id is not present in the canonical story ids
+                list of the topic.
+
         Returns:
             list. The updated list of canonical story ids.
         """
+        if story_id not in self.canonical_story_ids:
```

Comment:
should add similar check for add() methods, right? (I.e. don't want the item to already be in the list.)

Reply:
The validate() function checks for duplicates in all the lists. That would be enough, right?

Reply:
Done, added the check and tests for the same.